xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp

```diff
@@ -1553,8 +1553,19 @@ bool CAMLCodec::OpenDecoder(CDVDStreamInfo &hints)
   // handle video rate
   if (hints.fpsrate > 0 && hints.fpsscale != 0)
   {
-    // then ffmpeg avg_frame_rate next
-    am_private->video_rate = 0.5 + (float)UNIT_FREQ * hints.fpsscale / hints.fpsrate;
+    // ensure the hints yield a sensible refresh rate
+    if (hints.fpsrate / hints.fpsscale <= 120)
+    {
+      // then ffmpeg avg_frame_rate next
+      am_private->video_rate = 0.5 + (float)UNIT_FREQ * hints.fpsscale / hints.fpsrate;
```

Comment:
Does not know exactly how it behaves, but it is not interpreted as double and better placed to 0.5f here and below?

Seen only on other projects where `glm` becomes used that without "f" has generated errors with e.g. "float required" (I know no error possible here ;-) ), but made me think.

Reply:
This line of code has not changed as a result of my PR, I have just wrapped an if statement round it such that it only gets executed when sensible fps hints are passed along.